sqlitecluster/SQLiteClusterMessenger.cpp

```diff
@@ -42,21 +42,21 @@ bool SQLiteClusterMessenger::waitForReady(pollfd& fdspec, uint64_t timeoutTimest
         if (!result) {
             if (_shutDownBy) {
                 SINFO("[HTTPESC] Giving up because shutting down.");
-                return false;
+                return WaitForReadyResult::SHUTTING_DOWN;
             } else if (timeoutTimestamp && timeoutTimestamp < STimeNow()) {
                 SINFO("[HTTPESC] Timeout waiting for socket.");
-                return false;
+                return WaitForReadyResult::TIMEOUT;
             }
         } else if (result == 1) {
             if (fdspec.revents & POLLERR || fdspec.revents & POLLHUP || fdspec.revents & POLLNVAL) {
                 SINFO("[HTTPESC] Socket disconnected while waiting to be ready (" << type << ").");
-                return false;
+                return fdspec.events == POLLIN ? WaitForReadyResult::DISCONNECTED_IN : WaitForReadyResult::DISCONNECTED_OUT;
```

Comment:
What do `DISCONNECTED_IN` vs `DISCONNECTED_OUT` mean (or what does `POLLIN` mean)? Below it says this regarding `DISCONNECTED_OUT`:

> This is the case we're likely to get if the leader's port is closed.

Can you explain this or point me to where I can read more about it? (Maybe a comment would help)

Reply:
Discussed it 1:1 👍🏼